apps/checkout-app/pages/api/webhooks/mollie.ts

```diff
@@ -13,7 +13,13 @@ export default async function handler(
   res: NextApiResponse
 ) {
   if ("id" in req.body) {
-    const paymentData = await verifyPayment(req.body.id);
+    let paymentData;
+    try {
+      paymentData = await verifyPayment(req.body.id);
+    } catch (e) {
+      res.status(500).json({ error: "error while validating payment" });
+      return;
```

Comment:
Maybe you can rewrite it to avoid mutating variables

Reply:
I guess you could theoretically do something like this:

```ts
const paymentData = await verifyPayment(req.body.id).catch(() => {
    res.status(500).json({ error: "error while validating payment" });
});
```

To be honest, I don't like mixing `.then()` + `.catch()` with await. In this scenario we also cannot immediately send the response - the control flow will go to the `if (paymentData)` block`.

We can also use good old vars 😆 they have function scope, instead of block scope after all 😉
```js
    try {
      var paymentData = await verifyPayment(req.body.id);
    } catch (e) {
      res.status(500).json({ error: "error while validating payment" });
      return;
    }
```

Reply:
Suggested a solution here: https://github.com/saleor/saleor-checkout/pull/145#discussion_r924253885

Reply:
Thanks @mmiszy - really clean solution, I've changed it up a little bit (I'm not calling the function inside the utility, but rather await a promise) and named it `unpackPromise` - I dunno it just sounded nice in my head 😄